docs/core/task_library/contributing.md

```diff
@@ -0,0 +1,88 @@
+# Contributing
+
+Contributing to Prefect's task library is a great way to assist in open source development! Generally
+users have contributed tasks to the task library that accomplish a specific goal or action.
+
+There are a few key reasons why users would want to contribute tasks to the task library:
+
+- Gain experience contributing to an open source project
+- Increase adoption for libraries, tools, and frameworks by making an easy route for users of Prefect to
+interact with
+- Allow for tasks to _evolve_ with Prefect meaning that as paradigms and abstractions change in Prefect
+the task in the open source library will change with it
+- Open up collaboration to thousands of other developers who could use your task (they might fix bugs in
+the task you weren't aware of!)
+
+Not to mention that we occasionally also send Prefect swag to some of our open source contributors!
+
+<script>
+import { Tweet } from 'vue-tweet-embed/dist'
+
+export default {
+    components: {Tweet}
+}
+</script>
+
+<Tweet id="1298298873878847490"></Tweet>
+
+## Task Structure
+
+This snippet below is the general structure of a task contained in the task library. All you need to do
+is define the task's `__init__` and `run` functions! The `run` function is where any of your task's logic
+will live.
+
+```python
+from prefect import Task
+from prefect.utilities.tasks import defaults_from_attrs
+
+class YourTask(Task):
+    """
+    Task for doing something
+
+    Args:
+        - your_kwarg (string, optional): an optional kwarg
+        - **kwargs: additional keyword arguments to pass to the Task constructor
+    """
+
+    def __init__(
+        self,
+        your_kwarg: str = None,
+        **kwargs: Any
+    ):
+        self.your_kwarg = your_kwarg
+        super().__init__(**kwargs)
+
+    # Allows init kwargs to be passed to the run function if they are not overridden
```

Comment:
This isn't super clear to me - maybe
> Allows for certain attributes to optionally be provided at runtime, which will override the values provided at init

/shrug

Reply:
Going to actually expand on this outside of the code